src/connector.js

```diff
@@ -0,0 +1,612 @@
+		}
+	}
+	else if ( querySummaryState.hasError ) {
+		querySummaryElement.textContent = "";
+		querySummaryElement.innerHTML = resultErrorTemplateHTML;
+	}
+}
+
+// Update pagination
+function updatePagerState( newState ) {
+	pagerState = newState;
+	pagerElement.textContent = "";
+	let innerHTML = "";
+	if ( pagerState.hasPreviousPage ) {
+		const liNode = document.createElement( "li" );
+		liNode.innerHTML = previousPageTemplateHTML;
+		const aNode= liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => { 
+            pagerController.previousPage();
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );
+	}
+	pagerState.currentPages.forEach( ( page ) => {
+		const liNode = document.createElement( "li" );
+		if ( page == pagerState.currentPage ) {
+			liNode.setAttribute( "class","active" );
+		}
+		const pageNo = page;
+		liNode.innerHTML = pageTemplateHTML.replace( '%[page]', pageNo );
+		const aNode = liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => {
+            pagerController.selectPage( pageNo );
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );				
+	} );
+	if ( pagerState.hasNextPage ) {
+		const liNode = document.createElement( "li" );
+		liNode.innerHTML = nextPageTemplateHTML;
+		const aNode = liNode.querySelector( 'a' );
+		aNode.onclick = ( e ) => { 
+            pagerController.nextPage(); 
+            window.scrollTo( { top: 200, behavior: 'smooth' } );
+        };
+		pagerElement.appendChild( liNode );
+	}
+}
+
+// Subscribe to Headless controllers
+searchBoxController.subscribe( () => updateSearchBoxState( searchBoxController.state ) );
+resultListController.subscribe( () => updateResultListState( resultListController.state ) );
+querySummaryController.subscribe( () => updateQuerySummaryState( querySummaryController.state ) );
+pagerController.subscribe( () => updatePagerState( pagerController.state ) );
+
+// Listen to "Enter" key up event for search action
+if ( searchBoxElement ) {
+	searchBoxElement.onkeyup = ( e ) => {
+		lastCharKeyUp = e.keyCode;
+		if( e.keyCode === 13 ) {
+			e.preventDefault(); // Ensure it is only this code that runs
+			//searchBoxController.submit(); // not required, click on search button occurs automatically...
+		}
+		else {
+			searchBoxController.updateText( e.target.value );
+		}
+	};
+	searchBoxElement.onfocus = ( e ) => {
+		lastCharKeyUp = null;
+		searchBoxController.showSuggestions();
+	};
+}
+
+// Listen to click event for search action
+if (searchBoxButtonElement){
+	searchBoxButtonElement.onclick = ( e ) => {
+		if ( params.isAdvancedSearch ) {
+			return; // advanced search forces a post back
+        }
+		
+		e.preventDefault();
+
+		if ( searchBoxElement && searchBoxElement.value ) {
+			searchBoxController.submit();
+		}
+		else {
+			resultListElement.textContent = "";
+		}
+	};
+}
+
+// Remove Query suggestion if click elsewhere
```

Comment:
Additional requirement is applicable for popup, please check the WCAG and the WAI-ARIA Practice for more info.

Reply:
Will relay to @bblaisATcoveo for his subsequent PR, this will require some research, but is critical to be sorted out before going live. Thank you

Reply:
OPTIONAL for Go-live

Reply:
Point taken on the query suggestion "popup". Will close this change request since not expected to be used for go-live. If decision becomes that we want to use it, we still have a note to look into this.

Reply:
We have added a parameter called `unsupportedSuggestions` as a safety flag to allow for the auto-creation of the  suggestion element, allowing us to use it only under certain circumstances (demo, testing, etc.) while a more stable version is being developed.